lib/validate/net.go

```diff
@@ -36,11 +36,11 @@ func KubernetesSubnets(podCIDR, serviceCIDR string) error {
 				"invalid pod subnet: %q", podCIDR)
 		}
 
-		// the pod network should be /16 minimum so k8s can allocate /24 to each node
+		// the pod network should be /22 minimum so k8s can allocate /24 to each node (minimum 3 nodes)
 		ones, _ := podNet.Mask.Size()
-		if ones > 16 {
+		if ones > 22 {
```

Comment:
Just curious where this change is coming from. Has anyone tried/wanted to allocate smaller subnets for their clusters?

Reply:
It comes up with customer M from time to time. 

The problem is the pod cidr does need to be coordinated with the network to some degree, if pods within the cluster need to be able to connect to customer services. The way this tends to manifest, is the customer than allocates subnets for the pod/service cidr from their own subnet tracking, where a /16 is a quite large allocation. It does help that if they have multiple clusters they can re-use the /16 where some customers interpreted it as a /16 for each cluster(dev/staging/prod each get their own subnet), and also why we moved to the cgnat private IP range by default which is outside the RFC 1918 address space in use by most networks.